code/modules/surgery/organs/liver.dm

```diff
@@ -101,12 +101,76 @@
 			if(provide_pain_message && damage > 10 && DT_PROB(damage/6, delta_time)) //the higher the damage the higher the probability
 				to_chat(C, "<span class='warning'>You feel a dull pain in your abdomen.</span>")
 
-		else //for when our liver's failing
-			C.liver_failure(delta_time, times_fired)
 
 	if(damage > maxHealth)//cap liver damage
 		damage = maxHealth
 
+/obj/item/organ/liver/handle_failing_organs(delta_time)
+	if(HAS_TRAIT(src, TRAIT_STABLELIVER) || HAS_TRAIT(src, TRAIT_NOMETABOLISM))
+		return
+	return ..()
+
+/obj/item/organ/liver/organ_failure(delta_time)
+
+	switch(failure_time/LIVER_FAILURE_DIVIDER)
+		if(1)
+			to_chat(owner,"<span class='danger'>You feel stabbing pain in your abdomen!</danger>")
+		if(2)
+			to_chat(owner,"<span class='danger'>You feel a burning sensation in your gut!</danger>")
+			owner.vomit()
+		if(3)
+			to_chat(owner,"<span class='danger'>You feel painful acid in your throat!</danger>")
+			owner.vomit(blood = TRUE)
+		if(4)
+			to_chat(owner,"<span class='danger'>Overwhelming pain knocks you out!</danger>")
+			owner.vomit(blood = TRUE, distance = rand(1,2))
+			owner.emote("Scream")
+			owner.AdjustUnconscious(2.5 SECONDS)
+		if(5)
+			to_chat(owner,"<span class='danger'>You feel as if your guts are about to melt!</danger>")
+			owner.vomit(blood = TRUE,distance = rand(1,3))
+			owner.emote("Scream")
+			owner.AdjustUnconscious(5 SECONDS)
+
+	switch(failure_time)
+			//After 60 seconds we begin to feel the effects
+		if(1 * LIVER_FAILURE_DIVIDER to 2 * LIVER_FAILURE_DIVIDER - 1)
+			owner.adjustToxLoss(0.2 * delta_time)
+			owner.adjust_disgust(0.5 * delta_time)
+		if(2 * LIVER_FAILURE_DIVIDER to 3 * LIVER_FAILURE_DIVIDER - 1)
+			owner.adjustToxLoss(0.4 * delta_time)
+			owner.drowsyness += 1 * delta_time
+			owner.adjust_disgust(1 * delta_time)
+		if(3 * LIVER_FAILURE_DIVIDER to 4 * LIVER_FAILURE_DIVIDER - 1)
+			owner.adjustToxLoss(0.6 * delta_time)
+			owner.adjustOrganLoss(pick(ORGAN_SLOT_HEART,ORGAN_SLOT_LUNGS,ORGAN_SLOT_STOMACH,ORGAN_SLOT_EYES,ORGAN_SLOT_EARS),0.2 * delta_time)
+			owner.drowsyness += 2 * delta_time
+			owner.adjust_disgust(1.5 * delta_time)
+			if(DT_PROB(1.5, delta_time))
+				owner.emote("drool")
+		if(4 * LIVER_FAILURE_DIVIDER to INFINITY)
+			owner.adjustToxLoss(0.8 * delta_time)
+			owner.adjustOrganLoss(pick(ORGAN_SLOT_HEART,ORGAN_SLOT_LUNGS,ORGAN_SLOT_STOMACH,ORGAN_SLOT_EYES,ORGAN_SLOT_EARS),0.5 * delta_time)
+			owner.drowsyness += 3 * delta_time
+			owner.adjust_disgust(2 * delta_time)
+			if(DT_PROB(3, delta_time))
+				owner.emote("drool")
+
+/obj/item/organ/liver/on_owner_examine(datum/source, mob/user, list/examine_list)
+	if(!ishuman(user) || !(organ_flags & ORGAN_FAILING))
+		return
+
+	var/mob/living/carbon/human/humie = user
```

Comment:
Try to preserve the "user" part of the variable name, just knowing this is a human isn't as useful as knowing it's a human *user*.